### PR TITLE
Blog Subscription Widget: deprecate widget and transform to block

### PR DESCRIPTION
**Pull request by arcangelini**
#### Changes proposed in this Pull Request:

With the new current widget block editor, the Blog Subscription widget is redundant. The current Subscription Form block already exists and offers the same functions. This PR removes it from the `core/legacy-widget` and adds the widget to the REST API to allow a block `transforms:` to be created.

*This will preserve the following settings that exist in the block: show_subscribers_total, subscribe_button, subscribe_placeholder*.

#### Jetpack product discussion
Part of the widget => block project #21069 

#### Does this pull request change what data or activity we track or use?
Nope

#### Testing instructions:
* Before switching to this PR, go to Customizer ⇢ Widgets and add Legacy Widget choosing Blog Subscription from the dropdown. *Be sure to save changes*
* Next switch to this PR and view your customizer.
* You should see that Blog Subscription is no longer an option in the Legacy Widget block.
* Now click the previously added Legacy Blog Subscription and use the Transform button to change to the Subscription Form block.

### CODE REVIEW

**Comment by arcangelini** [comment]
The WPCOM test will fail due to the check against `widget_types_to_hide_from_legacy_widget_block`. After this PR is ready to merge I will [add the widget instance to the array here](https://code.a8c.com/diffusion/WP/browse/trunk/bin/tests/core/tests/blocks/block-editor.php$275).

Also, the Subscription Form block does not appear to have a section for a success message. Am I missing that somewhere, or is this not a setting available?

**Comment by jeherve** [comment]
> the Subscription Form block does not appear to have a section for a success message. Am I missing that somewhere, or is this not a setting available?

We did not implement that setting for the block yet. I logged this here:
https://github.com/Automattic/jetpack/issues/21187

**Comment by arcangelini** [comment]
Rebased and removed widget from Legacy block. Tested and everything is working great.

Markup 2021-09-28 at 15 02 47

**Review by jeherve — COMMENTED** [review]
This works well for me. Do you think we could also keep the widget title, in a heading block?

File: projects/plugins/jetpack/extensions/blocks/subscriptions/index.js [hunk 1]
```diff
@@ -58,6 +59,28 @@ export const settings = {
...
+				isMatch: ( { idBase, instance } ) => {
+					if ( ! instance?.raw ) {
+						return false;
+					}
+					return idBase === 'blog_subscription';
+				},
+				transform: ( { instance } ) => {
+					return createBlock( 'jetpack/subscriptions', {
+						showSubscribersTotal: instance.raw.show_subscribers_total,
+						submitButtonText: instance.raw.subscribe_button,
```
Comment (amustaque97) [inline projects/plugins/jetpack/extensions/blocks/subscriptions/index.js:77]: @arcangelini :wave: have you tried changing it to some different custom string! 
Example: `submitButtonText: 'Subscribe to my newsletter'`.
I tried something similar but it is not working for me somehow! :disappointed:  `submitButtonText` value is set to default value that you mentioned in the `attributes.js` file
Can you please try at your end and confirm?
Thanks

Comment (arcangelini) [reply projects/plugins/jetpack/extensions/blocks/subscriptions/index.js:77]: Yeah, if you look at my last screenshot it shows the placeholder text as "Placeholder" and the submit button as "Button".

I can test again though!

Comment (amustaque97) [reply projects/plugins/jetpack/extensions/blocks/subscriptions/index.js:77]: I'm trying to understand what this block of code does?
```
return createBlock( 'jetpack/subscriptions', {
    showSubscribersTotal: instance.raw.show_subscribers_total,
    submitButtonText: instance.raw.subscribe_button,
    subscribePlaceholder: instance.raw.subscribe_placeholder,
} );
```
If you see in the screenshot, I have commented default value of `submitButtonText` and if it has to read the value from `index.js` file `instance.raw.subscribe_button` then I have given some hard coded value `custom button value` which is not getting reflected. Button text is empty actually.

POC

Comment (amustaque97) [reply projects/plugins/jetpack/extensions/blocks/subscriptions/index.js:77]: @arcangelini, no need to test it again. It is working as expected. I understand now when that block of code is going to run.
Thank You :smile:

**Comment by arcangelini** [comment]
@jeherve I could, but the problem we have run into is that this heading block doesn't not get placed immediately above the subscription block. The only way to get it to do this would be using a Group block, but due to the way Gutenberg checks the transforms output we can not transform directly to a group block.

In the future, I think there will be a [Widget group block](https://github.com/WordPress/gutenberg/pull/34484) to transform to, but this has not gone live in core yet so I have opted to not use this.

This is what it looks like:

Markup 2021-10-10 at 00 39 28

**Review by sdixon194 — APPROVED** [review]

**Pull request merged**